### PR TITLE
support multiple pillow machines

### PR DESCRIPTION
**Pull request by emord**
Copying the format of celery processes, it's repetitive right now, but eventually that will change when they have differen processes

@dimagi/scale-team @millerdev

### CODE REVIEW

File: fab/services/templates/supervisor_pillowtop.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [program:{{ project }}-{{ environment }}-pillowtop-{{ pillow_name }}]
-command={{ virtualenv_current }}/bin/python {{ code_current }}/manage.py run_ptop {{ pillow_option }}
+command={{ virtualenv_current }}/bin/python {{ code_current }}/manage.py run_ptop {{ pillow_option }} --num_processes={{ num_processess }} --process_num={{ process_num }}
```
Comment (snopoke) [inline fab/services/templates/supervisor_pillowtop.conf:2]: where to these vars come from?

Comment (snopoke) [reply fab/services/templates/supervisor_pillowtop.conf:2]: never mind - I see it in the other PR

**Review by snopoke — APPROVED** [review]

**Comment by snopoke** [comment]
Might be good to add a line to docs somewhere explaining how to add / remove a pillow.

**Review by millerdev — APPROVED** [review]
Kinda hard to verify that the old `fab/pillows/*.yml` contents got transferred into new pillow configs in `fab/environments.yml`—assuming @emord has double-checked that.

**Comment by emord** [comment]
@millerdev yeah. that was the point of making htem alphabetical. when I went back through and checked I had missed a pillow.

@snopoke will add some docs

File: fab/environments.yml [hunk 2]
```diff
@@ -54,6 +54,58 @@ softlayer:
...
+      UnknownUsersPillow:
+        num_processes: 1
+      UserCacheInvalidatePillow:
+        num_processes: 1
+      UserGroupsDbKafkaPillow:
+        num_processes: 1
+      UserPillow:
+        num_processes: 1
+      XFormToElasticsearchPillow:
+        num_processes: 1
```
Comment (dannyroberts) [inline fab/environments.yml:108]: Maybe this is fine (not sure), but I'm wondering, does this mean that when a new pillow is added, it won't go into production (or softlayer, or icds) until it's also added here? A nice thing about the exclude/include stuff is that it made the default clear and small deviations from the default easy and clear in how they differed from it. (I get that that wasn't easily extensible to specifying which pillows go on which machines, though.)

**Review by dannyroberts — APPROVED** [review]
Didn't review in great detail, but the approach seems reasonable. Mostly I love that we're finally getting to this

File: fab/environments.yml (hunk 2 above)
Comment (emord) [reply fab/environments.yml:108]: Yeah. I had the same conflicted feelings about that, but need to get this out before pillow partitioning. I would be pro a refactor for both celery and pillows that enables including and excluding, but I think it's good that we're using the same type of config now

Comment (dannyroberts) [reply fab/environments.yml:108]: :+1:

**Pull request merged**